pandas/tests/indexing/multiindex/test_concat.py

```diff
@@ -0,0 +1,16 @@
+from pandas import (
+    DataFrame,
+    MultiIndex,
+    concat,
+)
+import pandas._testing as tm
+
+df1 = DataFrame({"col": ["a", "b", "c"]}, index=["1", "2", "2"])
+df2 = concat([df1], keys=["X"])
+
+iterables = [["X"], ["1", "2", "2"]]
+
+result = df2.index
+expected = MultiIndex.from_product(iterables)
```

Comment:
construct an expected dataframe and not just the index

use `tm.assert_frame_equal`